constraints_test.go

```diff
@@ -138,6 +140,23 @@ func TestConstraintCheck(t *testing.T) {
 		{"2", "2.1.1", true},
 		{"2.1", "2.1.1", true},
 		{"2.1", "2.2.1", false},
+		{"==2", "1", false},
+		{"==2", "3.4.5", false},
+		{"==2", "2.0.0", false},
+		{"==2", "2.0.0+alpha", false},
+		{"==2", "2.0.0-alpha", false},
+		{"==2", "2.0.1", false},
+		{"==2.1", "2.1.0", false},
+		{"==2.1.x", "2.1.0", false},
+		{"==2.1.x", "2.1.1", false},
```

Comment:
You can't have a version that's "2". So, a constraint of `==2` can never be matched against.

Since other comparisons allow shortened versions with assumptions on missing parts this looks to be confusing to user input that's run through here. That, I think, means more support requests.

node-semver does `==` and `===` which you can find out about at https://github.com/npm/node-semver?tab=readme-ov-file#comparison.

When I look at `===` with missing parts (e.g., `===1`) it falls into a loose rather than exact string situation. but, if you put a full semver in there is a string comparison.

What do you think of the node-semver model?

Reply:
I agree with your point, since there was no double equal operator I didn't thinked too much about it, maybe I am wrong, here what I think should be some constraints examples that follows the node-semver model:
- simple equal operator `=` :
```
"=2.x.x", "2.0.1", true
"=2.x.x", "2.0.1+mybuild", true
"=2.x.x", "2.0.1-beta+mybuild", true
"=2.x.x+build", "2.0.1", false
"=2.x.x+build", "2.0.1+mybuild", true
"=2.x.x-beta+build", "2.0.1+mybuild", (true or false)? # In this case, in the node model, the version and the build would correspond, so it could be equal if beta is not taken into account
"=2.x.x-beta+build", "2.0.1-beta+mybuild", true
```
- double equal operator `==` :
```
"==2.x.x", "2.0.1", true
"==2.x.x", "2.0.1+mybuild", false
"==2.x.x", "2.0.1-beta+mybuild", false
"==2.x.x+build", "2.0.1", false
"==2.x.x+build", "2.0.1+mybuild", true
"==2.x.x-beta+build", "2.0.1+mybuild", false
"==2.x.x-beta+build", "2.0.1-beta+mybuild", true
```
- triple equal operator `===` :
```
"===2.x.x", "2.0.1", false
"===2.x.x", "2.0.1+mybuild", false
"===2.x.x", "2.0.1-beta+mybuild", false
"===2.x.x+build", "2.0.1", false
"===2.x.x+build", "2.0.1+mybuild", false
"===2.x.x-beta+build", "2.0.1+mybuild", false
"===2.x.x-beta+build", "2.0.1-beta+mybuild", false
"===2.0.1", "2.0.1", true
"===2.0.1", "2.0.1+mybuild", false
"===2.0.1", "2.0.1-beta+mybuild", false
"===2.0.1+build", "2.0.1", false
"===2.0.1+build", "2.0.1+mybuild", true
"===2.0.1-beta+build", "2.0.1+mybuild", false
"===2.0.1-beta+build", "2.0.1-beta+mybuild", true
```

I'm pretty much ok with the node-semver model, and this example as well, if we can be sure to find a specific version or subset of versions it would be perfect 👍